dist/XSLoader/t/XSLoader.t

```diff
@@ -21,6 +16,103 @@ BEGIN {
     }
 }
 
+# mini test implementation. We're going to be playing with the XS bits of
+# various modules that may be used by Test::More, so it's best to avoid. Since
+# XSLoader is dual life, we can't use something like perl's t/test.pl
+my $planned_tests;
+my $tests;
+my $passed_tests;
+sub ok ($;$) {
+    my ($ok, $name) = @_;
+    $tests++;
+    $passed_tests += 1 if $ok;
+    print STDOUT "not "
+        if !$ok;
+    print STDOUT "ok $tests";
+    print STDOUT " - $name"
+        if defined $name;
+    print "\n";
+    return $ok;
+}
+sub is ($$;$) {
+    my ($got, $want, $name) = @_;
+
+    my $ok
```

Comment:
This makes my head hurt but totally works

Reply:
If you have a suggestion for an improvement, I'd welcome it. All the negatives do hurt the comprehensibility.

Reply:
Maybe this: Could be even clearer with more ifs etc...

```
   my $different = (defined $want xor defined $got)
                 || (defined $got && $got ne $want);
    my $ok = ! $different;
```

Reply:
I didn't love your proposal, but I've replaced this with something that should be clearer.

Reply:
Yes yours is great, thanks